### PR TITLE
Long quotes

### LINKED ISSUE #5881: Long quote posts  Expand /  Shrink (Feature Request)
Hi,

Can "Expand /  Shrink" feature be added to long quote posts?  as in the code box

### PR DESCRIPTION
**Pull request by MissAllSunday**
Fixes #5881

Adds a new setting for allowing users to expand/shrink a long quote.

Screenshot from 2020-02-23 19-06-03

Screenshot from 2020-02-23 19-57-51

### CODE REVIEW

**Comment by live627** [comment]
Can it be similar to the code tag in that it shows the first few lines while collapsed? (No scrollbar needed I think)

**Comment by MissAllSunday** [comment]
The code tag is just text which makes it easier to extract some of it and show a teaser when collapsed.  Unfortunately it is not that easy to do the same with a quoted post since it can have any number of DOM elements on it, making it a lot more difficult to reliable extract a part of it.

**Comment by live627** [comment]
Then use CSS max height and hidden overflow. Pretty sure that the code tag does that.

**Comment by MissAllSunday** [comment]
NVM overflow-y: hidden; will do the trick.

**Comment by MissAllSunday** [comment]
OK now theres a teaser
Screenshot from 2020-02-23 19-06-03


Theres only one issue with multiquote I'm still trying to see how to solve it.

**Comment by MissAllSunday** [comment]
It now gracefully handles multiquotes

Screenshot from 2020-02-23 19-57-51

**Comment by SychO9** [comment]
Since Antes is having some trouble in github, here's what he noticed while testing the PR:

> you get Shrink/Expand buttons even if the setting is set to 0

File: Themes/default/languages/Admin.english.php [hunk 1]
```diff
@@ -560,6 +560,8 @@
 $txt['edit_disable_time'] = 'Maximum time after posting to allow edit';
 $txt['preview_characters'] = 'Maximum length of last/first post preview';
 $txt['preview_characters_units'] = 'characters';
+$txt['quote_expand'] = 'Minimum quote height to add an expand link on large quotes';
+$txt['quote_expand_pixes_units'] = 'pixels';
```
Comment (Sesquipedalian) [inline Themes/default/languages/Admin.english.php]: Nit: please change this key to `quote_expand_pixels_units`. The `l` is missing.

**Review by Sesquipedalian — CHANGES_REQUESTED** [review]
LGTM, except one little thing. 😉

**Comment by gevv** [comment]
Hi,

Is this feature added? It doesn't work in the latest version.

**Comment by smfbigguy** [comment]
It works on my site fine.

**Comment by gevv** [comment]
For setting from the admin panel this is very good.

Thanks everyone.

**Pull request merged**